Mathlib/Analysis/Calculus/FirstDerivativeTest.lean

```diff
@@ -0,0 +1,237 @@
+"differentiable implies continuous",
+"an open interval is an open set", and "`fun x => -x` is antitone". -/
+
+
+/-- If `f` is differentiable on `(a,b)`, and `x ∈ (a,b)`, then `f` is differentiable at `x`.-/
+theorem differentiableOn_differentiableAt_Ioo.{u_1, u_2, u_3} {𝕜 : Type u_1}
+  [NontriviallyNormedField 𝕜] {E : Type u_2} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+    [LinearOrder E] [OrderClosedTopology E]
+    {a x b : E} (hab : x ∈ Set.Ioo a b)
+    {F : Type u_3} [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+    {f : E → F}
+    (hd₀ : DifferentiableOn 𝕜 f (Set.Ioo a b)) :
+    DifferentiableAt 𝕜 f x := by
+  apply DifferentiableOn.differentiableAt
+  exact hd₀
+  refine IsOpen.mem_nhds ?hs.hs hab
+  apply isOpen_Ioo
+
+/-- If `f` is continuous at `b` and differentiable on `(a,b)` then `f` is
+  continuous on the half-open interval `(a,b]`. -/
+theorem continuous_Ioc.{u_1, u_2, u_3} {𝕜 : Type u_1} [NontriviallyNormedField 𝕜]
+  {E : Type u_2} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+    [LinearOrder E] [OrderClosedTopology E]
+    {F : Type u_3} [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+    {f : E → F}
+    {a b : E}
+    (g₀ : a < b) (h : ContinuousAt f b)
+    (hd₀ : DifferentiableOn 𝕜 f (Set.Ioo a b)) : ContinuousOn f (Set.Ioc a b) := by
+  intro x hx
+  by_cases H : x = b
+  · subst H;simp_all;exact ContinuousAt.continuousWithinAt h
+  · have hab: x ∈ Set.Ioo a b := by
+      contrapose H
+      simp_all only [Set.mem_Ioo, and_imp, Set.mem_Ioc, true_and, not_lt, Decidable.not_not]
+      apply le_antisymm
+      tauto;tauto
+    have hD : DifferentiableAt 𝕜 f x := by
+      apply differentiableOn_differentiableAt_Ioo;repeat tauto
+    apply ContinuousAt.continuousWithinAt
+    exact DifferentiableAt.continuousAt hD
+
+/-- If `f` is continuous at `b` and differentiable on `(b,c)` then `f` is
+  continuous on the half-open interval `[b,c)`. -/
+theorem continuous_Ico.{u_1, u_2, u_3} {𝕜 : Type u_1} [NontriviallyNormedField 𝕜]
+  {E : Type u_2} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+    [LinearOrder E] [OrderClosedTopology E]
+    {F : Type u_3} [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+    {f : E → F}
+    {b c : E} (g₁ : b < c)
+    (h : ContinuousAt f b) (hd₁ : DifferentiableOn 𝕜 f (Set.Ioo b c)) :
+    ContinuousOn f (Set.Ico b c) := by
+  intro x hx
+  by_cases H : x = b
+  · subst H;simp_all;exact ContinuousAt.continuousWithinAt h
+  · have hab: x ∈ Set.Ioo b c := by
+      refine Set.mem_Ioo.mpr ?_;
+      constructor
+      · contrapose H; simp_all;
+        apply le_antisymm;tauto;tauto
+      · exact hx.2
+    have hD : DifferentiableAt 𝕜 f x := by
+      apply differentiableOn_differentiableAt_Ioo;repeat tauto
+    apply ContinuousAt.continuousWithinAt
+    · exact DifferentiableAt.continuousAt hD
+
+/-- If `f` is differentiable on a set `s` then so is `-f`. -/
+theorem differentiableOn_neg_Ioo
+  {f : ℝ → ℝ} {s : Set ℝ} (hd₀ : DifferentiableOn ℝ f s) :
+    DifferentiableOn ℝ (-f) s := by
```

Comment:
We have a generic `DifferentiableOn.neg`.

Reply:
Great! Using that, I was able to delete this "theorem" now.